trieste/acquisition/sampler.py

```diff
@@ -19,100 +19,77 @@
-    """
-
-    def __init__(self, sample_size: int, model: ProbabilisticModel):
-        """
-        :param sample_size: The desired number of samples.
-        :param model: The model to sample from.
-        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
-        """
-        tf.debugging.assert_positive(sample_size)
-
-        self._sample_size = sample_size
-        self._model = model
-
-    def __repr__(self) -> str:
-        """"""
-        return f"{self.__class__.__name__}({self._sample_size!r}, {self._model!r})"
-
-    @abstractmethod
-    def sample(self, at: TensorType) -> TensorType:
-        """
-        :param at: Input points that define the sampler.
-        :return: Samples.
-        """
-
-
-class ThompsonSampler(Sampler, ABC):
+class ThompsonSampler(ABC):
     r"""
     A :class:`ThompsonSampler` samples either the minimum values or minimisers of a function
     modeled by an underlying :class:`ProbabilisticModel` across a  discrete set of points.
 
     """
 
-    def __init__(self, sample_size: int, model: ProbabilisticModel, sample_min_value: bool = False):
+    def __init__(self, sample_min_value: bool = False):
         """
-        :param sample_size: The desired number of samples.
-        :param model: The model to sample from.
         :sample_min_value: If True then sample from the minimum value of the function,
             else sample the function's minimiser.
-        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
         """
-        super().__init__(sample_size, model)
         self._sample_min_value = sample_min_value
 
+    @property
+    def sample_min_value(self) -> bool:
+        return self._sample_min_value
+
     def __repr__(self) -> str:
         """"""
         return f"""{self.__class__.__name__}(
-        {self._sample_size!r},
-        {self._model!r},
         {self._sample_min_value})
         """
 
+    @abstractmethod
+    def sample(self, model: ProbabilisticModel, sample_size: int, at: TensorType) -> TensorType:
+        """
+        :param model: The model to sample from.
+        :param sample_size: The desired number of samples.
+        :param at: Input points that define the sampler.
+        :return: Samples.
+        """
+
 
 class ExactThompsonSampler(ThompsonSampler):
     r"""
     This sampler provides exact Thompson samples of the objective function's
-    maximiser :math:`x^*` over a discrete set of input locations.
+    minimiser :math:`x^*` over a discrete set of input locations.
 
     Although exact Thompson sampling is costly (incuring with an :math:`O(N^3)` complexity to
     sample over a set of `N` locations), this method can be used for any probabilistic model
     with a sampling method.
 
     """
 
-    def sample(self, at: TensorType) -> TensorType:
+    def sample(self, model: ProbabilisticModel, sample_size: int, at: TensorType) -> TensorType:
```

Comment:
why this change in arguments?

Reply:
So that you can specify a ThompsonSampler class without having access to the model it is going to be used on, i..e. when specifying a sampler as an argument when initializing rule or acquisition function.